### PR TITLE
[-] Technical - Use handlebars in lumber model generation

### PR DESCRIPTION
**Pull request by slimee**

### CODE REVIEW

**Comment by slimee** [comment]
[Lumber - Generate command creates hasMany relationships on NoSQL databases](https://trello.com/c/upSw1D9C/3871-lumber-generate-command-creates-hasmany-relationships-on-nosql-databases)

**Comment by slimee** [comment]
One field of the sequelize template is not tested: target/field attribute.
How to do that @VincentMolinie please?

**Comment by VincentMolinie** [comment]
> One field of the Sequelize template is not tested: ./target/field attribute.
> How to do that @VincentMolinie: having a `reference.targetKeyColumnUnconventional === true` in a Sequelize template.

To do an unconventional targetKey put everything in camelCase except the targetKey that you'll put in snake_case for example ;)

**Comment by slimee** [comment]
> > One field of the Sequelize template is not tested: ./target/field attribute.
> > How to do that @VincentMolinie: having a `reference.targetKeyColumnUnconventional === true` in a Sequelize template.
> 
> To do an unconventional targetKey put everything in camelCase except the targetKey that you'll put in snake_case for example ;)

Good, this a managed. 👍

File: services/dumper.js [hunk 1]
```diff
@@ -128,8 +129,11 @@ function Dumper(config) {
   }
 
   function writeModel(table, fields, references, options = {}) {
-    const templatePath = `${__dirname}/../templates/app/models/model.txt`;
-    const template = _.template(fs.readFileSync(templatePath, 'utf-8'));
+    const templatePath = config.dbDialect === 'mongodb' ?
+      `${__dirname}/../templates/app/models/mongo-model.hbs`
+      :
+      `${__dirname}/../templates/app/models/sequelize-model.hbs`;
```
Comment (VincentMolinie) [inline services/dumper.js]: We never use this syntax. The linter is not correctly set on this project.
It should have been 
```
? .....
: ....;
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@
 - Command Generate - Generate the forest routes to ease the use of `smart routes`.
 
 ### Changed
+- Command Generate - Use handlebars to generate models.
```
Comment (VincentMolinie) [inline CHANGELOG.md]: This was a technical change

Comment (VincentMolinie) [reply CHANGELOG.md]: It's missing a fix, you fixed the targetKey unconventionnal when in undersocred is false

File: templates/app/models/sequelize-model.hbs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    tableName: '{{table}}',{{#if underscored}}
+    underscored: true,{{/if}}{{#unless timestamps}}
+    timestamps: false,{{/unless}}{{#if schema}}
+    schema: process.env.DATABASE_SCHEMA,{{/if}}
+  });
+{{#if noId}}  {{modelName}}.removeAttribute('id');{{/if}}
+  // This section contains the relationships for this model. See: https://docs.forestadmin.com/documentation/v/v5/reference-guide/relationships#adding-relationships.
+  {{modelName}}.associate = (models) => {
+    {{#each references as |reference|}}
+    {{../modelName}}.belongsTo(models.{{reference.ref}}, {
```
Comment (VincentMolinie) [inline templates/app/models/sequelize-model.hbs]: [Misc] `../model` what is that ?

File: templates/app/models/sequelize-model.hbs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      foreignKey: {
+        name: '{{reference.foreignKeyName}}',{{#if reference.foreignKeyColumnUnconventional}}
+        field: '{{reference.foreignKey}}',{{/if}}
+      },{{#if reference.targetKey }}
```
Comment (VincentMolinie) [inline templates/app/models/sequelize-model.hbs]: Useless whitespace

File: templates/app/models/sequelize-model.hbs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      defaultValue: Sequelize.literal('{{{field.defaultValue}}}'),{{/if}}
+    },
+{{/each}}
+  }, {
+    tableName: '{{table}}',{{#if underscored}}
+    underscored: true,{{/if}}{{#unless timestamps}}
+    timestamps: false,{{/unless}}{{#if schema}}
+    schema: process.env.DATABASE_SCHEMA,{{/if}}
+  });
+{{#if noId}}  {{modelName}}.removeAttribute('id');{{/if}}
```
Comment (VincentMolinie) [inline templates/app/models/sequelize-model.hbs]: The indent is hard to read here

**Pull request merged**